drivers/at86rf2xx/at86rf2xx_getset.c

```diff
@@ -367,24 +447,64 @@ void at86rf2xx_set_cca_threshold(const at86rf2xx_t *dev, int8_t value)
     }
     /* transform the dBm value in the form
        that will fit in the AT86RF2XX_REG__CCA_THRES register */
-    value -= RSSI_BASE_VAL;
+    value -= at86rf2xx_rssi_base_values[dev->base.dev_type];
     value >>= 1;
     value &= AT86RF2XX_CCA_THRES_MASK__CCA_ED_THRES;
-    value |= AT86RF2XX_CCA_THRES_MASK__RSVD_HI_NIBBLE;
+    switch (dev->base.dev_type) {
+#if IS_USED(MODULE_AT86RF212B)
+        /* AT86RF212B has CCA_CS_THRES as bits [7; 4] in reg CCA_THRES,
+           so take care that we don´t override them. */
+        case AT86RF2XX_DEV_TYPE_AT86RF212B: {
+            uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__CCA_THRES);
+            tmp &= ~AT86RF2XX_CCA_THRES_MASK__CCA_ED_THRES;
+            value = tmp |= value;
+            break;
+        }
+#endif
+#if IS_USED(MODULE_AT86RF231)
+        /* AT86RF231 also has CCA_CS_THRES */
+        case AT86RF2XX_DEV_TYPE_AT86RF231: {
+            uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__CCA_THRES);
+            tmp &= ~AT86RF2XX_CCA_THRES_MASK__CCA_ED_THRES;
+            value = tmp |= value;
+            break;
+        }
+#endif
+#if IS_USED(MODULE_AT86RFA1)
+        /* AT86RFA1 also has CCA_CS_THRES */
+        case AT86RF2XX_DEV_TYPE_AT86RFA1: {
+            uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__CCA_THRES);
+            tmp &= ~AT86RF2XX_CCA_THRES_MASK__CCA_ED_THRES;
+            value = tmp |= value;
+            break;
+        }
+#endif
+#if IS_USED(MODULE_AT86RFR2)
+        /* AT86RFR2 also has CCA_CS_THRES */
+        case AT86RF2XX_DEV_TYPE_AT86RFR2: {
+            uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__CCA_THRES);
+            tmp &= ~AT86RF2XX_CCA_THRES_MASK__CCA_ED_THRES;
+            value = tmp |= value;
+            break;
+        }
+#endif
+    }
```

Comment:
The code is the same in all cases, so do you even need the switch?

Reply:
here is also a good candidate of `static inline` functions (e.g at86rf2_xxx`) that call the same function.

Reply:
E.g
```c
/* at86rf2xx.c */
void at86rf2xx_set_cca_threshold(const at86rf2xx_t *dev, int8_t value)
{
            uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__CCA_THRES);
            tmp &= ~AT86RF2XX_CCA_THRES_MASK__CCA_ED_THRES;
            value = tmp |= value;
}

/* at86rf231.h */
void at86rf231_set_cca_threshold(const at86rf2xx_t *dev, int8_t value)
{
    at86rf2xx_set_cca_threshold(dev, value);
}

/* at86rf233.h */
void at86rf233_set_cca_threshold(const at86rf2xx_t *dev, int8_t value)
{
    at86rf2xx_set_cca_threshold(dev, value);
}
```
Etc

Reply:
All types have `CCA_ED_THRES` in `CCA_THRESH` register. 
Inside the switch were types that also have CCA_CS_THRES.
But what is inside the cases would work for any kind.

So if it works for all kinds, a type specific static inline wrapper would not be necessary, right?
This should work for any.

 ```C
void at86rf2xx_set_cca_threshold(const at86rf2xx_t *dev, int8_t value)
{
    /* ensure the given value is negative, since a CCA threshold > 0 is
       just impossible: thus, any positive value given is considered
       to be the absolute value of the actually wanted threshold */
    if (value > 0) {
        value = -value;
    }
    /* transform the dBm value in the form
       that will fit in the AT86RF2XX_REG__CCA_THRES register */
    value -= at86rf2xx_rssi_base_values[dev->base.dev_type];
    value >>= 1;
    value &= AT86RF2XX_CCA_THRES_MASK__CCA_ED_THRES;

    uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__CCA_THRES);
    tmp &= ~AT86RF2XX_CCA_THRES_MASK__CCA_ED_THRES;
    value = tmp | value;
    value |= AT86RF2XX_CCA_THRES_MASK__RSVD_HI_NIBBLE; /* What is this? */
    at86rf2xx_reg_write(dev, AT86RF2XX_REG__CCA_THRES, value);
}
```
I could not find in any datasheet where
`AT86RF2XX_CCA_THRES_MASK__RSVD_HI_NIBBLE`
could come from.

Reply:
It makes sense to have this function as a common one.

Reply:
For this specific case, I think the conversion from dbm to internal values should be done outside of this function. Otherwise it removes all chances of caching the internal value and we would need to recalculate the internal value each time we call this function